pkg/utils/utils_test.go

```diff
@@ -18,9 +18,45 @@ func TestRedactClientID(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		actual := RedactClientID(test.clientID)
-		if actual != test.expected {
-			t.Fatalf("expected: %s, got %s", test.expected, actual)
-		}
+		t.Run(test.name, func(t *testing.T) {
+			actual := RedactClientID(test.clientID)
+			if actual != test.expected {
+				t.Fatalf("expected: %s, got %s", test.expected, actual)
+			}
+		})
+	}
+}
+
+func TestIsValidResourceID(t *testing.T) {
+	tests := []struct {
+		name        string
+		resourceID  string
+		expectedErr bool
+	}{
+		{
+			name:        "invalid resource id 0",
+			resourceID:  "invalidresid",
+			expectedErr: true,
+		},
+		{
+			name:        "invalid resource id 1",
+			resourceID:  "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourcegroups/0000/providers/Microsoft.ManagedIdentity/keyvault-identity-0",
```

Comment:
Good that we have test at this level, could we add it one level above as well since there is some logic there ?

Reply:
There is test one level above at MIC